services/admin/src/store.ts

```diff
@@ -360,23 +405,23 @@ export class Store {
         }
       }
 
-      this.client.run(
-        analyzeOnly ? 'rollback transaction' : 'commit transaction'
-      );
-      inTransaction = false;
-
       return ok({
         id,
         wasExistingFile: !!existing,
         newlyAdded,
         alreadyPresent,
       });
-    } catch (error) {
-      if (inTransaction) {
-        this.client.run('rollback transaction');
+    };
+
+    function shouldCommit(result: AddCastVoteResult) {
```

Comment:
Return type annotation?
```suggestion
    function shouldCommit(result: AddCastVoteResult): boolean {
```